src/Codeception/Lib/InnerBrowser.php

```diff
@@ -1063,11 +1063,19 @@ protected function getFormFor(Crawler $node)
      */
     protected function getFormValuesFor(Form $form)
     {
+        $formNodeCrawler = new Crawler($form->getFormNode());
         $values = [];
         $fields = $form->all();
         foreach ($fields as $field) {
-            if ($field instanceof FileFormField || $field->isDisabled() || !$field->hasValue()) {
-                continue;
+            if ($field instanceof FileFormField || $field->isDisabled()
+                ||
+                !$field->hasValue()
+            ) {
+                // if unchecked a checkboc and if there is hidden input with same name to submit unchecked value
+                $hiddenInput = $formNodeCrawler->filter('input[type=hidden][name="'.$field->getName().'"]:not([disabled])');
+                if (!count($hiddenInput)) {
+                    continue;
```

Comment:
This new code is in the wrong `if` block - it applies to file fields and disabled fields as well.

I think that it should be in its own block with `if(!$field->hasValue())` condition.  
I looked it up and `$field->hasValue()` returns false only for checkboxes and radio buttons [when they aren't selected](https://github.com/symfony/dom-crawler/blob/v5.2.4/Field/ChoiceFormField.php#L40-L53), so it is good for this purpose.

Also it would be good to get `$fieldValue` as a string right here to simplify the code below. The else case would set `$fieldValue = $field->getValue();` and the changes below would be simple assignments.